localstack/services/sqs/provider.py

```diff
@@ -1373,21 +1393,13 @@ def delete_message_batch(
     def purge_queue(self, context: RequestContext, queue_url: String) -> None:
         queue = self._resolve_queue(context, queue_url=queue_url)
 
-        with self._mutex:
-            # FIXME: use queue-specific locks
-            if queue.purge_in_progress:
-                raise PurgeQueueInProgress()
-            queue.purge_in_progress = True
-
-        # TODO: how do other methods behave when purge is in progress?
+        if config.SQS_DELAY_PURGE_RETRY and queue.purge_timestamp:
+            if (queue.purge_timestamp + 60) > time.time():
+                raise PurgeQueueInProgress(
+                    f"Only one PurgeQueue operation on {queue.name} is allowed every 60 seconds."
+                )
```

Comment:
all great points. i decided to simplify things a bit and remove the `purge_in_progress` all together, since in reality all the calls to `clear` will be extremely fast. i just put the mutex around the purge operation as you pointed out in the first comment.